nrfbsim: Enable and test GPIO(+TE) for 5340 and 54L15, and UARTE for 5340

aescolar opened the pull request:
Update nRF hw models to latest which include models of the GPIO & GPIOTE for the nrf5340 and nrf54l15, and UARTE for the nr5340, and enable the gpio and uart tests in these simulated platforms.

Also for the ci runtime uart tests we run in the babblesim job, enable the nrf5340.

Note: For the nrf54l15 some of the GPIO tests will fail at runtime (they will build fine => will not fail in CI) until the nordic nrfx hal is updated.

aescolar committed:
boards nrfbsim: Enable GPIO & GPIOTE peripherals for nrf54l15bsim

The HW models now support these pheripherals, let's enable them

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests gpio_basic_api: Enable for nrf54l15bsim

Enable this test, and provide overlays, in the nrf54l15bsim

Note this test specs 2 GPIO pins to be shorted. This can be done
for the simulation target by calling zephyr.exe with the option
`-gpio_conf_file=shorts_config.txt`
Where  that file would contain this one line (for the provided overlay)
---- shorts_config.txt
short 1.1 1.2
----

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests gpio_hogs: Enable for nrf54l15bsim

Enable this test, and provide overlays, in the nrf54l15bsim

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests gpio_get_direction: Enable for nrf54l15bsim

Enable this test by providing an overlay for the nrf54l15bsim

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
boards nrfbsim: Enable GPIO & GPIOTE peripherals for nrf5340bsim

The HW models now support these pheripherals, let's enable them

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests gpio_hogs: Enable for nrf5340bsim

Enable this test, and provide overlays, for the nrf5340bsim.
Both for the app and net core.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests gpio_basic_api: Enable for nrf5340bsim

Enable this test, and provide overlays, in the nrf5340bsim

Note this test specs 2 GPIO pins to be shorted. This can be done
for the simulation target by calling zephyr.exe with the option
`-gpio_conf_file=shorts_config.txt`
Where  that file would contain this one line (for the provided overlay),
for the cpunet:
---- shorts_config.txt
short 1.1 1.2
----
And this for the cpuapp:
---- shorts_config.txt
short 3.1 3.2
----

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests gpio_get_direction: Enable for nrf5340bsim

Enable this test by providing an overlay for the nrf5340bsim
Both for the app and net core.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
boards nrfbsim: Enable UART(E) peripherals for nrf5340bsim

The HW models now support this peripheral for these targets.
Let's enable them.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests uart_async_api: Enable in nrf5340bsim//cpuapp

Enable this test in the simulated nrf5340 by providing
an appropriate overlay.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
manifest: Update nRF hw models to latest

Update the HW models module to:
aeef3db9fa9e4b9d12b3bbec44f9cedc8fcb7d9c

Including the following:
aeef3db GPIO & GPIOTE: Improve notes and documentation
93549c4 UART: Fix command line parameter description
dbab746 nrf_hack: Add a few missing peripherals
4b1a61b 53 UARTE: Correct subscribe sideeffects function name
197e9cf docs: UART can be used now for 5340 in Zephyr
dc18d14 53: GPIO+TE: Add app core instances
b236c08 GPIO+TE: Add to 5340's netcore
7a621f6 GPIO: Clarify function description
1a2e1e4 GPIO+TE: Build for 54L
d121db4 nrf_gpio hal: Add new nrf_gpio_port_pin_{in,out}put_set
9bc41ce GPIOTE: Connect to DPPI
7141042 GPIOTE: Add simple support for PORT.SECURE & NONSECURE
dc930ba GPIOTE: Support having or not sense functionality per instance
a07180d GPIO: Support multiple GPIOTEs
14bdce5 GPIO: Collect status in single struct
b91da53 GPIOTE: Add support for N interrupt lines
9963424 GPIO & GPIOTE: Generalize to N instances
7379c23 GPIO+TE: Fix indentation
8d691c6 Add SPU registers stub
dd68ca9 Add NFCT register stub
cf2cc5b HW_models: NHW_NVMC: fix buffer read validation
f3db727 zephyr: module.yml: add `depends` field
09fc98f CMakeLists: remove trailing whitespace

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar commented:
CC @ilhanates

aescolar committed:
tests uart_pm: Enable for nrf5340bsim

Enable this test in the simulated nrf5340 and provide
an appropriate overlay.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

Thalley commented on tests/drivers/gpio/gpio_get_direction/boards/nrf54l15bsim_nrf54l15_cpuapp.overlay:
```diff
@@ -0,0 +1,7 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "nrf52_bsim.overlay"
```
Was not aware that overlays could include other overlays like this :)

Question is of course if it should. Looks a bit strange to me that the nrf54l15 cpuapp uses/includes the same overlay as the nrf52_bsim, especially when the `nrf52840dk_nrf52840.overlay` file doesn't

aescolar replied:
Doing this saves a lot of duplicate code. It can feel a bit funny to reuse overlays between SOCs, but lukily this is only configuring the GPIOs and what we change is common between them.
(Wrt. this particular test there is no overlay today for the nrf52840dk.)

aescolar committed:
doc boards nrfbsim: Mention the GPIO & GPIOTE as supported

Include in the list of supported peripherals the GPIO
and GPIOT for both the nrf5340 and nrf54l15

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests uart_mix_fifo_poll: Enable for nrf5340bsim

Enable this test in the simulated nrf5340 and provide
an appropriate overlay.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
doc boards nrfbsim: Mention the UARTE as supported for nrf5340

Include in the list of supported peripherals the UARTE for the
simulated nrf5340

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests/bsim: Also runtime test the UART drivers for the nrf5340

To increase coverage.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar commented:
@nordic-krch note the tests/drivers/uart/uart_async_api/ board overlay configuration enables CONFIG_UART_NRFX_UARTE_ENHANCED_RX as we discussed yesterday.
